### ISSUE TITLE
Feature Request: Temporary disable/override for gamepad mappings

### ISSUE THREAD

**Issue by mikey-dg** [WootingKb/wootility-issues#129]
**Is your feature request related to a problem? Please describe.**
Right now, the only way to temporarily disable gamepad mappings (including analog axes) and/or re-enable keys blocked by Gamepad Mapping Override is to switch to the digital profile, or another profile without gamepad mappings.  Switching profiles seems faster and more reliable in the latest firmware version (2.4.6 on the Wooting Two HE), especially when switching to the digital profile, but it can still take a few seconds before key input is properly accepted when switching to an analog profile.

This isn't particularly ideal for games where both analog input and normal key typing are equally important and sometimes mutually exclusive-- e.g. the game continues to accept controller input for character or UI actions when typing in a text input box or navigating a dialog or menu with keyboard input, causing undesirable outcomes.  Gamepad Mapping Override only addresses half of this issue if the game requires frequent normal keyboard input, and the only solutions are sub-optimal: mapping gamepad functions to whatever few (if any) keys the game doesn't use in text input, menus or dialogs, or constantly switching profiles and waiting out the initialization delay on the analog profile.

**Describe the solution you'd like**
The simplest solution would be a mappable key that can instantly switch off all controller output and enable keys overriden by Gamepad Mapping Override while held.  A additional mappable key that does this as a toggle and has an indicator color might be preferable for some users.

**Describe alternatives or workarounds you've considered**
I attempted to remap affected keys on the function layer, hoping that Fn would at least bypass the Gamepad Mapping Override, but this isn't the case - Fn has no effect on a gamepad-mapped key with the override enabled.

The reverse works slightly better: omitting the gamepad-mapped keys on the main layer mapping in lieu of using Gamepad Mapping Override, and then using a function layer mapping that includes them at least allows selective digital input of those keys, but it's an unwieldy and high-effort solution, and still doesn't solve the issue of unwanted gamepad input while trying to type.

**Comment by Rocky04** [comment]
Hmm, such things may make everything more complex. The option to disable the digital keys where an analog binding is present is already a quality of live feature. Following that road would lead to other similar things to address the then new introduced side effects.

**The recommended approach is still to just disable the digital keys in the game. By that you don't need to switch the profile at all and you are still be able to navigate in the menu or type in text in the chat or console.**

While I see a benefit is adding a key to toggle the override feature I don't such that for toggling the analog bindings.

**Comment by mikey-dg** [comment]
The benefit to toggling the analog bindings and controller buttons is entirely limited to older games that handle joystick/controller input somewhat poorly.  I've encountered a few older Windows games and plenty of DOS games that continue to accept controller input while menus are up or text boxes are active, and many of those games (particularly the DOS games) don't allow you to rebind their digital keys at all.

The latter is a scenario that Gamepad Mapping Override helps with, but only when you can bind analog inputs over the top of a game's existing movement keys.  When the game also requires a lot of typing, you're forced to bind all of your analog inputs to e.g. the numpad, which isn't really a great solution for a game where you need more than a dozen keys available nearby, or where you can't rebind keys at all and all of the game's important keys are on the main keyboard area.  The former problem, of the game always accepting analog input or controller buttons even when doing text/menu actions, currently also has no solution other than profile switching or sub-optimal binding location.

Basically, this is a proposed solution to having to bind analog inputs/controller buttons to the numpad any time a game doesn't handle them well on the letter keys.  It's also a way to temporarily regain use of the numpad in games where that is otherwise an optimal binding location (e.g. old Windows/DOS space and flight sims).

**Comment by mikey-dg** [comment]
> The benefit to toggling the analog bindings and controller buttons is entirely limited to older games that handle joystick/controller input somewhat poorly.

After trying a few other games, just like to amend this, and note that an instant toggle for analog/digital keys would also greatly help games that have good analog input with vehicles and poor analog input for character movement (e.g. GTA, Battlefield and Unreal Tournament games).  The analog input 'works' in these games in all scenarios, but it's poor enough when moving your player character and important enough with vehicles that an instant toggle would be immensely helpful.  While in some cases it's partly the fault of the game's analog movement implementation, in others (such as UT) the speed and responsiveness of a digital input is simply required outside of vehicle movement.

I've seen a few otherwise glowing reviews of the 2 HE that note this problem with this type of game, and also lament the delay in switching profiles every time you need to get in and out of a vehicle.  This feature (especially as a toggle) would be a much better solution to this issue.

**Comment by 0nlyDev** [comment]
I would like to see this happen as well

**Comment by skadlig** [comment]
This would be a nice feature in X4: Foundation but I'm guessing also in other space flight games like Star citizen. 

In X4 I'm using wasd + qe for strafing plus rolling with analog gamepad controls. I also have a need though for the keyboard to write names of stations, search for wares to trade or find certain types of ships. These kinds of games also tend to have a big heap of different key that have a need to map where an extra layer of keys would be valuable.

**Comment by BigBrainAFK** [comment]
The ideal solution to this is the same as we suggest for GTA5. Have 2 profiles. One with gamepad mappings and one without. GTA5 needs it as the vehicle and onfoot controls are completely different. It would be the same here though

**Comment by HikenMC** [comment]
2026 and I still want this feature as a new 80he user. My use-case is Trackmania, a racing game where both analog input and digital have their benefits.
I wish to be able to use my function layer to enable analog gamepad input so on my main layer it's digital.

**Comment by mikey-dg** [comment]
I agree, there's still a need for this.

While profile switching speed has further improved, there's still over a second of keyboard-wide downtime that's rather irritating when you're trying to play a game that requires frequent and timely switches between analog and digital inputs.

A toggle key with an indicator color that immediately suspends analog axis/gamepad button output and allows digital key presses through would significantly help.

A momentary delay when re-enabling analog output wouldn't be a significant problem, as disabling (in my experience) is typically the more time-sensitive action.  Even if there's a short delay, any method of toggling gamepad output that can improve on the current profile switch time and that doesn't disable the entire keyboard when switching would be a nice quality of life enhancement.
